igor-web/src/main/java/com/netflix/spinnaker/igor/config/GoogleCloudBuildProperties.java

```diff
@@ -33,22 +40,37 @@
     return this.accounts.stream().map(BuildService::getView).collect(Collectors.toList());
   }
 
-  @Data
-  public static class Account implements BuildService {
-    private String name;
-    private String project;
-    private String subscriptionName;
-    private String jsonKey;
-    private Permissions.Builder permissions = new Permissions.Builder();
+  @NonnullByDefault
+  @Value
+  public static final class Account implements BuildService {
+    private final String name;
+    private final String project;
+    private final String subscriptionName;
+    private final String jsonKey;
+    private final Permissions permissions;
 
-    @Override
-    public BuildServiceProvider getBuildServiceProvider() {
-      return BuildServiceProvider.GCB;
+    @Builder
+    @ConstructorBinding
+    @ParametersAreNullableByDefault
+    public Account(
+        String name,
+        String project,
+        String subscriptionName,
+        String jsonKey,
+        Permissions.Builder permissions) {
+      this.name = Strings.nullToEmpty(name);
```

Comment:
similar question to the other PR: are any of these fields required? should there be a constraint check rather than a nullToEmpty?

Reply:
Good point, addressed in #742.